static/app/components/modals/widgetBuilder/addToDashboardModal.tsx

```diff
@@ -69,8 +69,22 @@ function AddToDashboardModal({
   const [selectedDashboardId, setSelectedDashboardId] = useState<string | null>(null);
 
   useEffect(() => {
-    fetchDashboards(api, organization.slug).then(setDashboards);
-  }, []);
+    // Track mounted state so we dont call setState on unmounted components
+    let unmounted = false;
+
+    fetchDashboards(api, organization.slug).then(response => {
+      // If component has unmounted, dont set state
+      if (unmounted) {
+        return;
+      }
+
+      setDashboards(response);
+    });
+
+    return () => {
```

Comment:
Just a question from someone new around here:

It seems cumbersome to need to remember this when you're making fetch requests. Are there any existing hooks that handle this logic for you? Or have there been any TSC discussions around making fetch requests better?

Reply:
Yes, it definitely is. Frontend would greatly benefit from something like react query (or similar) - I encourage you to bring it up and I think it really should be up to the core team to work on topics like that :)